lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -1180,6 +1180,31 @@ LogicalResult OrOp::canonicalize(OrOp op, PatternRewriter &rewriter) {
     return success();
   }
 
+  // or(mux(c_1, a, 0), mux(c_2, a, 0), ..., mux(c_n, a, 0)) -> mux(or(c_1, c_2,
+  // .., c_n), a, 0)
+  if (auto firstMux = op.getOperand(0).getDefiningOp<comb::MuxOp>()) {
+    APInt value;
+    SmallVector<Value> conditions;
+    auto check = [&](Value v) {
+      auto mux = v.getDefiningOp<comb::MuxOp>();
+      if (!mux)
+        return false;
+      conditions.push_back(mux.getCond());
+      return firstMux.getTwoState() &&
+             firstMux.getTrueValue() == mux.getTrueValue() &&
+             matchPattern(mux.getFalseValue(), m_ConstantInt(&value)) &&
+             value.isZero();
```

Comment:
firstMux.getTwoState is examined multiple times. Is this supposed to be mux.getTwoState?

Also, it should be sufficient to check that the false values are all the same as opposed to specializing this for zero.

Reply:
> Also, it should be sufficient to check that the false values are all the same as opposed to specializing this for zero.

I had same thought but I don't think this works for non-zero false values-- `or(mux(c_1, a, 2), mux(c_2, a, 2))` != `mux(or(c_1, c_2), a, 2)`.  If only `c_1` or `c_2` are true, result should be `or(a, 2)` not `a`.